### PR TITLE
ELPP-3180 Implemented IntDNS$n entries for EC2

### PR DESCRIPTION
**Pull request by giorgiosironi**
Useful for clusters of nodes that want to talk to each other. For example, nodes that are follower can connect to `end2end--search--1.elife.internal`, the leader node of the cluster.

### CODE REVIEW

**Comment by giorgiosironi** [comment]
Going to try to apply this first step on `search--end2end` (1 ELB, 1 node, DNS entries for both the ELB and the node behind it).

File: src/buildercore/cfngen.py [hunk 1]
```diff
@@ -407,7 +407,12 @@ def _title_has_been_updated(title, section):
     def legacy_title(title):
         # some titles like EC2Instance1 were originally EC2Instance
         # however, no reason not to let EC2Instance2 be created?
-        return title.strip("1")
+        if title == 'EC2Instance1':
+            return 'EC2Instance'
+        if title == 'ExtraStorage1':
+            return 'ExtraStorage'
+        if title == 'MountPoint1':
+            return 'MountPoint'
```
Comment (seanwiseman) [inline src/buildercore/cfngen.py]: Could be 
```
def legacy_title(title):
    if title in ['EC2Instance1', 'ExtraStorage1', 'MountPoint1']:
        return title.strip('1')
```

**Comment by giorgiosironi** [comment]
Update on end2end applied correctly.

**Comment by giorgiosironi** [comment]
Running https://alfred.elifesciences.org/job/test-search/228 and then I'll apply to prod. It's still only 1 node but behind the load balancer.

**Comment by giorgiosironi** [comment]
Applied also to `search--prod`.

**Comment by giorgiosironi** [comment]
Added, but
```
PHP Fatal error:  Uncaught JMS\Serializer\Exception\InvalidArgumentException: The cache directory "/srv/search/src/Search/../../var/cache" is not writable. in /srv/search/vendor/jms/serializer/src/JMS/Serializer/SerializerBuilder.php:153
Stack trace:
```
some problem for the formula to solve.

**Comment by giorgiosironi** [comment]
https://github.com/elifesciences/builder-base-formula/pull/82

**Comment by giorgiosironi** [comment]
Search can't take advantage of multiple nodes yet, this PR is ready as-is.

**Pull request merged**